documents/invoice.md

```diff
@@ -0,0 +1,529 @@
+  "customer_details": {
+    "id": "cust_E7q0trFqXgExmT",
+    "name": "Gaurav Kumar",
+    "email": "gaurav.kumar@example.com",
+    "contact": "9999999999",
+    "gstin": null,
+    "billing_address": {
+      "id": "addr_E7q0ttqh4SGhAC",
+      "type": "billing_address",
+      "primary": true,
+      "line1": "Ground & 1st Floor, SJR Cyber Laskar",
+      "line2": "Hosur Road",
+      "zipcode": "560068",
+      "city": "Bengaluru",
+      "state": "Karnataka",
+      "country": "in"
+    },
+    "shipping_address": {
+      "id": "addr_E7q0ttKwVA1h2V",
+      "type": "shipping_address",
+      "primary": true,
+      "line1": "Ground & 1st Floor, SJR Cyber Laskar",
+      "line2": "Hosur Road",
+      "zipcode": "560068",
+      "city": "Bengaluru",
+      "state": "Karnataka",
+      "country": "in"
+    },
+    "customer_name": "Gaurav Kumar",
+    "customer_email": "gaurav.kumar@example.com",
+    "customer_contact": "9999999999"
+  },
+  "order_id": "order_E7q0tvRpC0WJwg",
+  "line_items": [
+    {
+      "id": "li_E7q0tuPNg84VbZ",
+      "item_id": null,
+      "ref_id": null,
+      "ref_type": null,
+      "name": "Master Cloud Computing in 30 Days",
+      "description": "Book by Ravena Ravenclaw",
+      "amount": 399,
+      "unit_amount": 399,
+      "gross_amount": 399,
+      "tax_amount": 0,
+      "taxable_amount": 399,
+      "net_amount": 399,
+      "currency": "INR",
+      "type": "invoice",
+      "tax_inclusive": false,
+      "hsn_code": null,
+      "sac_code": null,
+      "tax_rate": null,
+      "unit": null,
+      "quantity": 1,
+      "taxes": []
+    }
+  ],
+  "payment_id": null,
+  "status": "issued",
+  "expire_by": 1589765167,
+  "issued_at": 1579765167,
+  "paid_at": null,
+  "cancelled_at": null,
+  "expired_at": null,
+  "sms_status": "pending",
+  "email_status": "pending",
+  "date": 1579765167,
+  "terms": null,
+  "partial_payment": true,
+  "gross_amount": 399,
+  "tax_amount": 0,
+  "taxable_amount": 399,
+  "amount": 399,
+  "amount_paid": 0,
+  "amount_due": 399,
+  "currency": "INR",
+  "currency_symbol": "₹",
+  "description": "Invoice for the month of January 2020",
+  "notes": [],
+  "comment": null,
+  "short_url": "https://rzp.io/i/2wxV8Xs",
+  "view_less": true,
+  "billing_start": null,
+  "billing_end": null,
+  "type": "invoice",
+  "group_taxes_discounts": false,
+  "created_at": 1579765167
+}
+```
+-------------------------------------------------------------------------------------------------------
+
+### Update invoice
+
+```go
+line_items := make(map[string]interface{})
+line_items["0"] = map[string]interface{}{
+      "id": "li_JKqakYLuD5qxvi",
+      "name": "Book / English August - Updated name and quantity",
+      "quantity": 1,
+    }
+
+data:= map[string]interface{}{
+  "line_items": line_items,
+  "notes": map[string]interface{}{
+    "updated-key": "An updated note.",
+  },
+}
+body, err := client.Invoice.Invoice.Edit("<invoiceId>", data, nil)
```

Comment:
```suggestion
body, err := client.Invoice.Invoice.Update("<invoiceId>", data, nil)
```
nit: Better method name, which even matches the operation Update.

Reply:
done